examples/nextjs/services/nacelle.js

```diff
@@ -23,42 +33,103 @@ client.data.collectionPage = (params) => clientPIM.data.collectionPage(params);
 client.data.allCollections = (params) => clientPIM.data.allCollections(params);
 
 // PREVIEW MODE SETUP
-const previewBranchName = 'preview';
-const vercelBranch = process.env.VERCEL_GITHUB_COMMIT_REF;
-const inPreviewBranch = vercelBranch && vercelBranch === previewBranchName;
-const previewModeToggled =
-  process.env.NACELLE_PREVIEW_MODE === 'true' || inPreviewBranch;
-const previewVariablesExist = Boolean(
-  process.env.CONTENTFUL_PREVIEW_SPACE_ID &&
-    process.env.CONTENTFUL_PREVIEW_API_TOKEN
+const previewModeToggled = process.env.NACELLE_PREVIEW_MODE === 'true';
+
+const pimPreviewVariablesExist = Boolean(
+  process.env.SHOPIFY_STOREFRONT_ACCESS_TOKEN && process.env.MYSHOPIFY_DOMAIN
+);
+const cmsPreviewVariablesExist = Boolean(
+  process.env.CONTENTFUL_SPACE_ID && process.env.CONTENTFUL_PREVIEW_API_TOKEN
 );
 
-if (previewModeToggled && !previewVariablesExist) {
-  console.warn(
-    '\nPreview Mode not activated due to missing environment variables:\n' +
-      (!process.env.CONTENTFUL_PREVIEW_SPACE_ID &&
-        '- CONTENTFUL_PREVIEW_SPACE_ID\n') +
-      (!process.env.CONTENTFUL_PREVIEW_API_TOKEN &&
-        '- CONTENTFUL_PREVIEW_API_TOKEN\n')
-  );
-}
+if (previewModeToggled) {
+  if (!pimPreviewVariablesExist || !cmsPreviewVariablesExist) {
+    warnMissingEnvVars([
+      'SHOPIFY_STOREFRONT_ACCESS_TOKEN',
+      'MYSHOPIFY_DOMAIN',
+      'CONTENTFUL_SPACE_ID',
+      'CONTENTFUL_PREVIEW_API_TOKEN'
+    ]);
+  } else {
+    // Initialize the Shopify Preview Connector
+    const shopifyConnector = new NacelleShopifyPreviewConnector({
+      shopifyApiKey: process.env.SHOPIFY_STOREFRONT_ACCESS_TOKEN,
+      shopifyDomain: process.env.MYSHOPIFY_DOMAIN.split('.').shift()
+    });
+
+    client.data.product = (params) =>
```

Comment:
This could probably be done with something like `Object.keys(client.data).reduce(() => {}, {})`, right? Nitpick.

Reply:
It could. But I'm trying to optimize for "ease-of-grokking" for the merchant dev who filed [`NS-539`](https://nacelle.atlassian.net/browse/NS-539).